nexus/src/nexus.rs

```diff
@@ -2798,7 +2835,12 @@ impl Nexus {
         })
     }
 
-    pub async fn updates_refresh_metadata(&self) -> Result<(), Error> {
+    pub async fn updates_refresh_metadata(
+        &self,
+        opctx: &OpContext,
+    ) -> Result<(), Error> {
+        opctx.authorize(authz::Action::Modify, &authz::FLEET).await?;
```

Comment:
CC @iliana I put this check here to avoid querying the tuf repo unless the user is authorized.  The database queries are separately authorized in datastore.rs.

Reply:
I think this works for now. In the future, this code path will be called within Nexus when automatic updates are enabled (perhaps once every few hours, or some other regular cadence). What might this look like when we start doing that?

Reply:
For that, we could create an `OpContext` when Nexus starts up specifically for use to do automatic updates, similar to what we do for saga recovery:
https://github.com/oxidecomputer/omicron/blob/c66d2f6d47823a3b38b6b983a0380866e97ef362/nexus/src/nexus.rs#L211-L216
This uses `OpContext::for_background`:
https://github.com/oxidecomputer/omicron/blob/c66d2f6d47823a3b38b6b983a0380866e97ef362/nexus/src/context.rs#L349

We could create a separate built-in user for that context that would only have whatever privileges are needed for it.  Then whenever we did any of these operations, we'd use that background `OpContext` that's associated with that user.